esahg/esahg.php

```diff
@@ -1,149 +1,170 @@
-				} elseif ( is_file( $file ) ) {
-					if ( false === strstr( $file, 'wp-content/plugins/' ) && false === strstr( $file, 'wp-content/themes/' ) ) {
-						$files[] = $file;
-					}
-				}
-			}
-		}
-		closedir( $handle );
-		return $files;
-	} else {
-		echo '<p>' . sprintf( __( 'Apologies, it seems the zip file has disappeared, please manually download it to %s or try again.', 'esahg' ), ESAHGPATH );
-	}
-} // end walk_wp_directory()
-
-function hash_files( $files = array() ) {
-	if ( empty( $files ) ) { return false; }
-
-	global $download_to;
-	sort( $files );
-	$file_hashes = '<?php' . "\r\n" . '$filehashes = array(' . "\r\n";
-	foreach ( $files as $file ) {
-		$file_hashes .= "'" . str_replace( $download_to . '/wordpress/', '', $file ) . "' => '" . md5( $file ) . "',\r\n";
-	}
-	$file_hashes .= ");\r\n?>";
-
-	return $file_hashes;
+<?php
+/*
+Plugin Name: Exploit Scanner Auto Hash Generator
+Plugin URI: https://github.com/mattyrob
+Description: Checks for and then automatically generates hash files for use in the Exploit Scanner plugin for WordPress.
+Version: 1.0b
+Author: Matthew Robinson
+Author URI: https://github.com/mattyrob
+Licence: GPL3
+Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=2387904
+Text Domain: esahg
+*/
+
+define( 'ESAHGDIR', trailingslashit( plugin_basename( dirname( __FILE__ ) ) ) );
+define( 'ESAHGPATH', WP_CONTENT_DIR . '/' . ESAHGDIR );
+add_action( 'admin_menu', 'esahg_admin' );
+
+function esahg_admin() {
+	load_plugin_textdomain( 'eashg', 'wp-content/plugins/' . ESAHGDIR, '/' . ESAHGDIR );
+	add_management_page( __( 'Exploit Scanner Auto Hash Generator', 'eashg' ), __( 'Exploit Scanner Auto Hash Generator', 'eashg' ), apply_filters( 'eashg_capability', 'manage_options' ), 'eashg', 'eashg_menu' );
+} // end eashg_admin()
+
+function eashg_menu() {
+	global $wpdb;
+	echo '<div class="wrap">';
+	echo '<h2>' . __( 'Exploit Scanner Automatic Hash File Generator', 'esahg' ) . '</h2>';
+	if ( ! is_plugin_active( 'exploit-scanner/exploit-scanner.php' ) ) {
+		$exit_msg = sprintf( __( '%s is not active on this site, please download and activate it.', 'esahg' ), '<a href="https://wordpress.org/plugins/exploit-scanner/">Exploit Scanner</a>' );
+		exit( $exit_msg );
+	}
+	require( ABSPATH . WPINC . '/version.php' );
+	$es_path = WP_CONTENT_DIR . '/plugins/exploit-scanner/';
+	$hash_filename = 'hashes-' . $wp_version . '.php';
+	$file = $es_path . $hash_filename;
+
+	// instantiate WordPress file system for later
+	$creds = request_filesystem_credentials( site_url() . '/wp-admin/', '', false, false, array() );
+	if ( ! WP_Filesystem( $creds ) ) {
+		echo '<p>' . __( 'Unable to enable the WordPress Filesystem.', 'esahg' ) . '</p>';
+		exit;
+	}
+	global $wp_filesystem;
+
+	if ( ! is_readable( $file ) ) {
+		// attempt to get hashes from Wordpress.org API
+		$response = wp_remote_get( 'https://api.wordpress.org/core/checksums/1.0/?version=' . $GLOBALS['wp_version'] );

```

Comment:
Ideally, the URL here would be wrapped in `esc_url()` to ensure it's safe to pass to `wp_remote_get()`. It's also preferable to use `wp_safe_remote_get()`